paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/playground/CustomEndpointDialog.kt

```diff
@@ -0,0 +1,104 @@
+package com.stripe.android.paymentsheet.example.playground
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+
+@Composable
+internal fun CustomEndpointDialog(
+    currentUrl: String?,
+    onDismiss: () -> Unit,
+    onConfirm: (String?) -> Unit
+) {
+    var urlField by remember { mutableStateOf(currentUrl ?: "") }
+
+    AlertDialog(
+        onDismissRequest = { onDismiss() },
+        title = {
+            Text(
+                text = "Enter custom endpoint url",
+                style = MaterialTheme.typography.h6
+            )
+        },
+        text = {
+            Column {
+                ValidatedTextField(
+                    label = "Backend url",
+                    value = urlField,
+                    validator = {
+                        when {
+                            it.startsWith("http").not() -> "Must be a valid url"
+                            it.endsWith("/").not() -> "Must end with /"
+                            else -> null
+                        }
+                    },
+                    onValueChange = { urlField = it }
+                )
+                TextButton(
+                    onClick = { onConfirm(null) }
+                ) {
+                    Text("Reset to default")
+                }
```

Comment:
Nit: I think we can add replace the `Cancel` button with this. We don’t need a cancel button, as you can just click outside the dialog.